### PR TITLE
Support MoFQ4 on arc

### PR DESCRIPTION
**Pull request by cyita**
## Description

Support Mixture of Formats Quantization on arc.

### 1. Why the change?

Select the optimal 4-bit format on a layer-wise basis. 

### 2. User API changes

```python
    model = AutoModelForCausalLM.from_pretrained(model_path,
                                                 load_in_low_bit="mixed_4bit",
                                                 optimize_model=True,
                                                 trust_remote_code=True,
                                                 use_cache=True)
```

### CODE REVIEW

File: python/llm/src/bigdl/llm/transformers/model.py [hunk 1]
```diff
@@ -111,7 +111,7 @@ def load_convert(cls, q_k, optimize_model, *args, **kwargs):
         invalidInputError(q_k in ggml_tensor_qtype,
                           f"Unknown load_in_low_bit value: {q_k}, expected:"
                           f" sym_int4, asym_int4, sym_int5, asym_int5, sym_int8, nf3, nf4, "
-                          "fp4, fp8 or fp16.")
+                          "fp4, fp8, fp16 or mofq4.")
```
Comment (jason-dai) [inline python/llm/src/bigdl/llm/transformers/model.py:114]: change to "mixed_4bit"

File: python/llm/src/bigdl/llm/ggml/quantize.py [hunk 2]
```diff
@@ -34,7 +34,8 @@
                      "nf3": 11,
                      "fp16": 12,
                      "fp8": 15,
-                     "fp4": 16}
+                     "fp4": 16,
+                     "mofq4": 17}     # Mixture of Formats Quantization 4 bits
```
Comment (jason-dai) [inline python/llm/src/bigdl/llm/ggml/quantize.py:38]: change to "mixed_4bit"

**Comment by cyita** [comment]
| task       | mixed_4bit | q4_0  | fp4   |
|------------|------------|-------|-------|
| arc        | 60.67      | 60.67 | 59.90 |
| truthfulqa | 50.91      | 50.98 | 50.02 |

**Review by rnwang04 — APPROVED** [review]
LGTM

**Comment by cyita** [comment]
@hkvision

**Pull request merged**